stream/src/main/scala/org/apache/pekko/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -686,6 +686,15 @@ private final case class SavedIslandData(
  */
 @InternalApi private[pekko] object GraphStageTag extends IslandTag
 
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[pekko] object GraphStageIsland {
+  // used as a type hint when creating the array of logics
+  private final val logicArrayType = Array.empty[GraphStageLogic]
```

Comment:
`Array` is mutable, so doesn't this mean everyone shares the same `logics`? Wouldn't that be problematic? (though it's interesting that the tests succeed)

Reply:
You can't, it's size 0

Reply:
This seems safe enough because it is an empty array. Can we rename it as emptyArray? I would still be very surprised if this improves performance. The hotspot compiler should quickly deal with this if is a hotspot.